certbot/main.py

```diff
@@ -215,6 +215,10 @@ def _find_cert(config, domains, certname):
               RenewableCert instance or None.
     """
     action, lineage = _find_lineage_for_domains_and_certname(config, domains, certname)
+    if action == "renew":
```

Comment:
@ohemorange that logic was added because it greatly improved user experience; I don't think removing it is a good idea.

I agree it's good to minimise state, but not at the expense of functionality.